hashintel/hash
# Add styles for Select Menus and Chips

## 🌟 What is the purpose of this PR?



This PR adds styling for Select, Menu and Chips components. It also includes examples (which is a huge chunk of this PR) of how to achieve different variations of these components

## ⚠️ Known issues


- Icons other than the delete icon can't be placed on the right side of the Chip component
- There are some keyboard navigation issues in Select Menus with a TextField
- MenuItem doesn't currently handle links. To have links in a menu, we have to do something like 
```
<Link href="/" noLinkStyle>
    <MenuItem />
</Link>
```
This should be made simpler by just doing 
```
  <MenuItem href="/" />
```
- `MenuItem` needs a faded prop for MenuItems that have a slightly lighter color than others. Currently we pass in styles via `sx` prop to achieve this.
- Error UI for Select has not been implemented.
- For MenuItems with a checkbox, the entire MenuItem has a blue background when selected. The background should be transparent since the checkbox's checked state depicts if it's selected or not

## 🐾 Next steps


- Address known issues in a follow up [task](https://app.asana.com/0/1202055186269230/1202146804390145). Most of this changes will require creating a custom MenuItem and Select component.

## 🔍 What does this change?




- Added a custom `Chip` component which wraps over MUI's Chip component and adds the following props to MUI Chip's existing props. 
   - *hasCircleStartIcon* -> makes it possible to achieve this [design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=841%3A72304).
   - *rectangular* -> reduces the border radius of the chip
- Added a `ChipGroup` component which makes it possible to implement [compound chips](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=841%3A72304)
- Updated styling of OutlinedInput. This helps fix a ui issue with `Select` since it uses it under the hood
- Updated styles of `ListItemText`, `ListItemSecondaryAction`, `ListItemIcon`, `ListItemAvatar` when rendered as children to `MenuItem`. These components are used to achieve different `MenuItem` variations when combined.
- Refactor header dropdowns (account, notification and actions dropdowns) to use Menu/MenuItem setup.
- Overrode the default Select styles to match our design system.
- Add examples of Select Menus and Chips usage in playground.
- Breaks down playground page into subsections
- Updated eslint `no-restricted-imports` for new custom components


### 📜 Does this require a change to the docs?



- No

## 🔗 Related links





- [Add styles for Selection Menus](https://app.asana.com/0/1202055186269230/1201926880408096) _(internal)_

## 🛡 What tests cover this?


- Existing tests


## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Go to `/playground`
1.  Confirm components there match the designs in figma ([select menus](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=831%3A67739), [chips](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=841%3A72304))

## 📹 Demo


![Screenshot 2022-04-19 at 11 03 42](https://user-images.githubusercontent.com/16928688/164010400-0bd9dcef-8030-4e00-8946-e7e76076bb8a.png)

![Screenshot 2022-04-19 at 14 04 48](https://user-images.githubusercontent.com/16928688/164010644-9139431f-7d8f-4f7a-abd1-0c41c012c2c6.png)

![Screenshot 2022-04-19 at 14 14 19](https://user-images.githubusercontent.com/16928688/164012249-e1228348-2fb6-4b9a-bc04-a054e865af82.png)
